examples/autosharding/statefulset.yaml

```diff
@@ -10,11 +11,13 @@ spec:
   replicas: 2
   selector:
     matchLabels:
+      app.kubernetes.io/component: exporter
```

Comment:
I believe that `matchLabels` in a Deployment object is immutable. Could you please verify whether we can seamlessly upgrade from the latest KSM release to this version?

Reply:
You are right. `The StatefulSet "kube-state-metrics" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', 'updateStrategy' and 'minReadySeconds' are forbidden`

For such scenarios we delete and recreate object when updates are impossible

Would that mean we cannot new labels in future to avoid this?

Reply:
Only `matchLabels` is immutable but we can add new labels to the underlying pods. So we can just remove the label from here but keep it everywhere else. As long as `matchLabels` are a subset of the pod labels, everything should work as expected.

Reply:
Looks like I need to update https://github.com/kubernetes/kube-state-metrics/blob/master/jsonnet/kube-state-metrics/kube-state-metrics.libsonnet#L292 so matchLabels will only take existing labels

```jsonnet
        selector: { matchLabels: ksm.podLabels },
```

to

```jsonnet
        selector: { matchLabels: ksm. commonLabels },
```

Reply:
Seems like it. We need to use different label sets for pod labels and selector